### DISCUSSION TITLE
get error

### DISCUSSION THREAD

**Discussion by elhadiabdallah** [spatie/laravel-pdf#25] [Q&A]
i use this code from docs 
   ```php
 return pdf()
    ->view('pdf.invoice', compact('invoice'))
    ->name('invoice-2023-04-10.pdf')
    ->download();
```
i get this error 
Spatie\LaravelPdf\PdfBuilder::name(): Argument #1 ($downloadName) must be of type string, null given, called in /home/abdallahelhadi/learning/dashy/vendor/spatie/laravel-pdf/src/PdfBuilder.php on line 136

when i used argument in download method like name of file, i get this error 
The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/home/abdallahelhadi/learning/dashy/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/1150465337-0970458001704473191\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false}}'" failed. Exit Code: 127(Command not found) Working directory: /home/abdallahelhadi/learning/dashy/public Output: ================ Error Output: ================ sh: 1: npm: not found sh: 1: node: not found

**Comment by ArielMejiaDev** [comment]
The compact should be inside the view method as second param, view method has two Paraná, the first is a string to identify the blade file and the second param for the data, that is passed to that view

**Reply by elhadiabdallah** [reply]
I know this, but what about the error

**Comment by its4nas** [comment]
I think The ->name() method you used is not a valid method in Laravel if you are generating PDFs using the barryvdh/laravel-dompdf package.

### Try this 
``` 
return PDF::loadView('pdf.invoice', compact('invoice'))->download('invoice-2023-04-10.pdf');
